index.d.ts

```diff
@@ -0,0 +1,123 @@
+declare module "fetch-ponyfill" {
+    export default FetchPonyfill.bootstrap;
+}
+
+declare namespace FetchPonyfill {
+    export function bootstrap(options?: PonyfillCreateOptions): FetchPonyfillStatic;
+
+    interface PonyfillCreateOptions {
+        Promise?: Function;
+        XMLHttpRequest?: Function;
+    }
+
+    interface FetchPonyfillStatic {
+        fetch(input: RequestInfo, init?: RequestInit): Promise<Response>;
+        Headers: typeof Headers;
+        Request: typeof Request;
+        Response: typeof Response;
+    }
+
+    type HeadersInit = Headers | string[][] | { [key: string]: string };
```

Comment:
Pretty much everything starting from here is copied over from @types/whatwg-fetch